shark_turbine/kernel/wave/utils.py

```diff
@@ -305,3 +305,14 @@ def compile_and_invoke(
             **benchmark_flags,
         )
         _print_bench_result(benchmark_results, bench_file)
+
+
+def safe_subs(input: Any, subs: List[Tuple[IndexExpr, IndexExpr]]) -> Any:
+    """
+    Substitute input using provided `subs` list if input is sympy object.
+    Otherwise return input unchanged.
+    """
```

Comment:
Nice! Should we instantiate the indexing context within this function so you don't need to pass subs or you can make subs optional? That will avoid having to instantiate an indexing context multiple times in the code.

Reply:
Done